nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -150,12 +150,16 @@ LifecycleManager::isActiveCallback(
 }
 
 void
-LifecycleManager::CreateActiveDiagnostic(diagnostic_updater::DiagnosticStatusWrapper & stat)
+LifecycleManager::CreateDiagnostic(diagnostic_updater::DiagnosticStatusWrapper & stat)
 {
-  if (system_active_) {
-    stat.summary(diagnostic_msgs::msg::DiagnosticStatus::OK, "Nav2 is active");
-  } else {
-    stat.summary(diagnostic_msgs::msg::DiagnosticStatus::ERROR, "Nav2 is inactive");
+  if (state_of_managed_nodes_ == NodeState::ACTIVE || state_of_managed_nodes_ == NodeState::PAUSED) {
+    stat.summary(diagnostic_msgs::msg::DiagnosticStatus::OK, "Managed nodes' status is OK");
+  } else if (state_of_managed_nodes_ == NodeState::UNKNOWN){
+    stat.summary(diagnostic_msgs::msg::DiagnosticStatus::ERROR, "An error has occurred during node state transition");
+  } else if (state_of_managed_nodes_ == NodeState::UNCONFIGURED){
+    stat.summary(diagnostic_msgs::msg::DiagnosticStatus::WARN, "Managed nodes have not been configured");
```

Comment:
Question, not a request (yet) for a change: Is it really a warning that its just not setup yet? 

@mikeferguson I'd appreciate your opinion here if you had a few moments

Reply:
My rationale was simply that it might be useful to get warning if the nodes have not been configured (and additionally since it was an error before users might expect some yelling from diagnostics), but I can see it being just a normal state. Don't have strong feelings on it one way or another.

Reply:
Lets make it a normal state. Its not totally unheard of for folks to have many operating modes for certain nodes (e.g. swapping from GPS to lidar-localization when going from outdoor to indoor) which are implemented by transitioning up / down the appropriate items. I think its not an warning state. Lets go ahead and change that

Reply:
Also changed the if-else chain into a switch case.